parsing/location.ml

```diff
@@ -442,17 +460,18 @@ let () =
     (function
       | Sys_error msg ->
           Some (errorf ~loc:(in_file !input_name)
-                "I/O error: %s" msg)
+                ("I/O error: %s"[@i18n]) msg)
```

Comment:
I believe that attributes should be used when the semantics of the code is not modified, and that extensions are to be used when the semantics of the code cannot be understood by ignoring the annotation. Shouldn't this be an extension?

Reply:
This attribute is here to help extraction tools à la xgettext; it is only here to indicate that this string should be part of the translation table exposed to translators. Since it has no effect in the code, I believe that an attribute is the right choice.

Reply:
Oh, I thought this elaborated into application of some `I18n` function. Thanks for the clarification!

Reply:
To make these annotation a little clearer, I added some basic documentation for them in `i18n.mli`.

Reply:
When I looked at the code I wondered why you didn't use `[@i18n]` instead of `I18n.s` by default.  It would seem to me to be more consistent (and possibly less ugly).

Reply:
Using directly `I18n.s` is not possible since the argument is a format string, however it would be perfectly possible to add a `I18n.f: ('a,'b,'c,'d) format4 -> (`a,`b,`c`d) I18n.f` function that plays the role of the current `[@i18n]` annotation and enforce the fact that the format string argument of `Location.errorf` and `Location.raise_errorf` needs to be marked as localizable.

Reply:
Sure, I understand there is a problem in this case, but I was wondering why you've used `I18n.s` in lots of other places (where there are just normal string literals) instead of the attribute.  It seems inconsistent.

Reply:
The function `I18n.s` is used to translate user-facing strings as soon as possible with the aim to coax the typechecker into enforcing that localized strings and non-localized ones are not mixed together. I can see some inconsistencies here, but I would rather remove the use of attributes alltogether than `I18n.s`.